onepwd/plugins/action/update_s3_values_of_item.py

```diff
@@ -141,17 +141,11 @@ def run(self, tmp=None, task_vars=None, **kwargs):
 
         # Update Secret if changes are present
         if (check_bucket and check_key and check_secret and check_endpoint_url) == False and overwrite == True:
-            if SECRET_NAME == "nextcloud":
-                onepwd.OnePwd.update_s3_values_of_standard_s3_item(op, title=SECRET_NAME, vault=vault, BUCKET_NAME=BUCKET_NAME, ACCESS_KEY=ACCESS_KEY, ACCESS_SECRET=ACCESS_SECRET, ENDPOINT_URL=ENDPOINT_URL)
-            elif SECRET_NAME == "server":
+            if SECRET_NAME == "server":
                 onepwd.OnePwd.update_s3_values_of_server_item(op, title=SECRET_NAME, vault=vault, BUCKET_NAME=BUCKET_NAME, ACCESS_KEY=ACCESS_KEY, ACCESS_SECRET=ACCESS_SECRET, ENDPOINT_URL=ENDPOINT_URL)
-            elif SECRET_NAME == "ionos-s3-password-backup":
-                onepwd.OnePwd.update_s3_values_of_standard_s3_item(op, title=SECRET_NAME, vault=vault, BUCKET_NAME=BUCKET_NAME, ACCESS_KEY=ACCESS_KEY, ACCESS_SECRET=ACCESS_SECRET, ENDPOINT_URL=ENDPOINT_URL)
-            elif SECRET_NAME == "nextcloud-ionos-s3-password-backup":
-                onepwd.OnePwd.update_s3_values_of_standard_s3_item(op, title=SECRET_NAME, vault=vault, BUCKET_NAME=BUCKET_NAME, ACCESS_KEY=ACCESS_KEY, ACCESS_SECRET=ACCESS_SECRET, ENDPOINT_URL=ENDPOINT_URL)
             else:
-                raise Exception("Secret is neither the server, ionos-s3-password-backup  nor the nextcloud secret. Upload function for any other secret not implemented yet")
-            print("Secret updated...")
+                onepwd.OnePwd.update_s3_values_of_standard_s3_item(op, title=SECRET_NAME, vault=vault, BUCKET_NAME=BUCKET_NAME, ACCESS_KEY=ACCESS_KEY, ACCESS_SECRET=ACCESS_SECRET, ENDPOINT_URL=ENDPOINT_URL)
```

Comment:
The `update_s3_values_of_standard_s3_item` method of [__init__.py](https://github.com/hpi-schul-cloud/infra-tools/blob/OPS-3484-nextcloud-s3-backup-refactoring/onepwd/src/onepwd/__init__.py#L127-L134) that gets called here looks not quite right. Is it really on purpose that `ACCESS_KEY` changes `ACCESS_SECRET`, `ACCESS_SECRET` changes `BUCKET_NAME` and `BUCKET_NAME` changes `ACCESS_KEY`? If it really is intended, how comes?

Reply:
works independently of the order :)